pkg/kube/client.go

```diff
@@ -291,6 +293,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 		return nil
 	})
 
+	if err != nil || len(updateErrors) != 0 {
+		for _, info := range newlyCreatedResources {
+			kind := info.Mapping.GroupVersionKind.Kind
+
+			c.Log("Deleting newly created %s with the name %q in %s...", kind, info.Name, info.Namespace)
+			if err := deleteResource(info); err != nil {
+				return fmt.Errorf("Failed to delete newly created %s with the name %q in %s: %s", kind, info.Name, info.Namespace, err)
```

Comment:
this should probably be added to `updateErrors` instead of returning; we lose all context behind what failed to upgrade.

Reply:
Fixed by adding cleanupErrors, because there are 2 cases:

1. Some newly added resources has been created successfully. And some new resource failed. In this case there will be `err != nil`.
2. All newly added resources has been created successfully. And there are errors in update stage. In this case `len(updateErrors) != 0`.

`cleanupErrors` will be appended either to the `err` or `updateErrors`.